src/assets/js/subsection-highlighting.js

```diff
@@ -0,0 +1,27 @@
+function highlightSubnav(targetUrl) {
+    function activateTarget() {
+        document.querySelectorAll(".Docs-subnav-link").forEach(item => {
+          if (item.href == targetUrl) {
+            item.classList.add("is-active");
+            accordionPanel = item.parentNode.parentNode
+            sectionHeaderButton = accordionPanel.parentNode.firstElementChild.firstChild
+            if (sectionHeaderButton.type == "button" && !!accordionPanel.getAttribute("aria-hidden")) {
+                sectionHeaderButton.click()
+            }
+          } else {
+            item.classList.remove("is-active");
+          }
+        });
+    }
+    
+    // If the active item is not one we clicked, close it first.
+    let activeItem = document.querySelector(".Docs-subnav-link.is-active");
+    if (activeItem && activeItem.href != targetUrl) {
+        activeItem.classList.remove("is-active");
+        setTimeout(activateTarget, 300);
+    } else {
+        activateTarget();
+    }
+}
+
+highlightSubnav(document.URL)
```

Comment:
i think it'd be good to wrap this call in an iife, just to make sure the timing is okay when it runs. you can follow the example of defining a listener and attaching it to the `readystatechange` document event in `link.js`. something like: 
```js
(function () {
  if (document.readyState !== "complete") {
    return
   }

  function eventListener () {
    highlightSubnav(document.URL)
    document.removeEventListener('readystatechange', eventListener)
  }

  document.addEventListener('readystatechange', eventListener);
})();

```

Reply:
ah yes, I was going to do that actually and it slipped my mind. Thanks for calling this out!

Reply:
This code won't work I think, because if the document is not ready we're returning and don't attach the eventlistener to the readystate.
I implemented this instead, it should do the trick, I think

```
(function () {
  function eventListener () {
    highlightSubnav(document.URL)
    document.removeEventListener('readystatechange', eventListener)
  }
  document.addEventListener('readystatechange', eventListener);
})();
```